File: app/src/main/java/com/lttrung/dormitory/ui/viewprofile/ViewProfileUseCase.java
package com.lttrung.dormitory.ui.viewprofile;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Completable;

@Singleton
public interface ViewProfileUseCase {
    // Bổ sung các thuộc tính có thể edit
    Completable editProfile();
}

File: app/src/main/java/com/lttrung/dormitory/ui/main/viewprofile/ViewProfileUseCase.kt
package com.lttrung.dormitory.ui.main.viewprofile

import com.lttrung.dormitory.database.data.network.models.UserProfile
import io.reactivex.rxjava3.core.Single
import javax.inject.Singleton

@Singleton
interface ViewProfileUseCase {
    fun getUserProfile(): Single<UserProfile>
}